### PR TITLE
add wait-for condition for telegraf

### PR DESCRIPTION
**Pull request by Thrinadh-Kumpatla**
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -93,6 +93,29 @@ Return the appropriate apiVersion for statefulset.
...
+{{- else -}}
+{{.Values.telegraf.enabled }}
+{{- end -}}
+{{- end -}}
+
+{{- define "statsd.url" -}}
+{{- if .Values.telegraf_sidecar.enabled -}}
+{{- printf "localhost:8125" }}
+{{- else -}}
+{{include "telegraf.fullname" .}}:{{.Values.telegraf.service.port }}
```
Comment (AkhileshK8y) [inline templates/_helpers.tpl:116]: This function is not present.. Please change it...

Comment (AkhileshK8y) [inline templates/_helpers.tpl:116]: This function is not present.. Please change it...

File: templates/_helpers.tpl [hunk 2]
```diff
@@ -93,6 +93,29 @@ Return the appropriate apiVersion for statefulset.
...
+{{- end -}}
+{{- end -}}
+
+{{- define "statsd.enabled" -}}
+{{- if .Values.telegraf_sidecar.enabled -}}
+true
+{{- else -}}
+{{.Values.telegraf.enabled }}
+{{- end -}}
+{{- end -}}
```
Comment (AkhileshK8y) [inline templates/_helpers.tpl:110]: we are prone to get nil pointer exception in statsd enabled function..

Comment (AkhileshK8y) [inline templates/_helpers.tpl:110]: we are prone to get nil pointer exception in statsd enabled function..

**Comment by saurav-malani** [comment]
I tried testing this change:

To test if rudder-server actually waits until telegraf sidecar is ready or not. I simply made this change:
`telegraf_sidecar:
  enabled: false`
  with the expectation that `telegraf` should never come up & since this PR makes rudder-server wait for `telegraf` to come up. It should also never come up. But, this change throws following error: 
  `Error: INSTALLATION FAILED: template: rudderstack/templates/statefulset.yaml:112:81: executing "rudderstack/templates/statefulset.yaml" at <include "statsd.enabled" .>: error calling include: template: rudderstack/templates/_helpers.tpl:108:9: executing "statsd.enabled" at <.Values.telegraf.enabled>: nil pointer evaluating interface {}.enabled`

**Comment by saurav-malani** [comment]
I tested the change, it seems to be working fine. Previous error is fixed. Also, `wait-for` condition seems to be working fine.

**Review by AkhileshK8y — APPROVED** [review]
Lgtm

**Review by saurav-malani — APPROVED** [review]
I tested with the new change, and everything seems to be working fine as per expectation.

**Pull request merged**